sycl/test/e2e_test_requirements/no-unsupported-without-info.cpp

```diff
@@ -0,0 +1,548 @@
+// This test is intended to ensure that we have no tests marked as
+// UNSUPPORTED without an information added to a test.
+// For more info see: sycl/test-e2e/README.md
+//
+// The format we check is:
+// UNSUPPORTED: lit,features
+// UNSUPPORTED-TRACKER: [GitHub issue URL|Internal tracker ID]
+// *OR*
+// UNSUPPORTED: lit,features
+// UNSUPPORTED-INTENDED: explanation why the test isn't intended to be run with this feature
+//
+// GitHub issue URL format:
+//     https://github.com/owner/repo/issues/12345
+//
+// Internal tracker ID format:
+//     PROJECT-123456
+//
+// REQUIRES: linux
+//
+// Explanation of the command:
+// - search for all "UNSUPPORTED" occurrences, display line with match and the next one
+//   -I, --include to drop binary files and other unrelated files
+// - in the result, search for "UNSUPPORTED" again, but invert the result - this
+//   allows us to get the line *after* UNSUPPORTED
+// - in those lines, check that UNSUPPORTED-TRACKER or UNSUPPORTED-INTENDED is
+//   present and correct. Once again, invert the search to get all "bad" lines
```

Comment:
i didnt actually review the code but i can say it looks correct to run in check-sycl, thanks!